### PR TITLE
{storage} `az storage blob metadata/copy/service-properties`: Migrate to track2

### PR DESCRIPTION
**Pull request by bebound**
**Related command**
`az storage blob metadata` `az storage blob copy` `az storage blob service-properties` `az storage blob service-properties delete-policy`

**Description**
- Migrate below commands to use track2 SDK:
`az storage blob metadata show/update`
`az storage blob copy cancel/start-batch`
`az storage blob service-properties show/update`
`az storage blob service-properties delete-policy show/update`

- `az storage blob copy start-batch` will skip empty folder when use `--source-container` rather than raise `InvalidInput` error. This logic has already been implemented when use `--source-share`.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
storage

File: src/azure-cli/azure/cli/command_modules/storage/operations/blob.py [hunk 1]
```diff
@@ -307,11 +304,66 @@ def set_service_properties(client, parameters, delete_retention=None, delete_ret
     return client.get_blob_service_properties()
 
 
-def storage_blob_copy_batch(cmd, client, source_client, container_name=None,
-                            destination_path=None, source_container=None, source_share=None,
-                            source_sas=None, pattern=None, dryrun=False):
-    """Copy a group of blob or files to a blob container."""
+# pylint: disable=too-few-public-methods
+class BlobServiceProperties:
```
Comment (calvinhzy) [inline src/azure-cli/azure/cli/command_modules/storage/operations/blob.py:308]: Is there a corresponding class in SDK?

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 2]
```diff
@@ -969,6 +969,21 @@
 short-summary: Manage blob metadata.
 """
 
+helps['storage blob metadata show'] = """
+type: command
+short-summary: Returns all user-defined metadata for the specified blob or snapshot.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:974]:
```suggestion
short-summary: Return all user-defined metadata for the specified blob or snapshot.
```

File: src/azure-cli/azure/cli/command_modules/storage/operations/blob.py (hunk 1 above)
Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/storage/operations/blob.py:308]: `get_service_properties()` returns a dict and I need to create a generic update command. It seems that generic update command only accept instance as input, so I convert it to `BlobServiceProperties`.

File: src/azure-cli/azure/cli/command_modules/storage/util.py [hunk 3]
```diff
@@ -16,6 +16,19 @@ def collect_blobs(blob_service, container, pattern=None):
...
 
+def collect_blobs_track2(blob_service, container, pattern=None):
+    """
+    List the blobs in the given blob container, filter the blob by comparing their path to the given pattern.
+    Return files only.
+    """
+    blobs = collect_blob_objects(blob_service, container, pattern)
+    r = []
+    for name, blob in blobs:
+        if blob.content_settings.content_md5 is not None:
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/util.py:27]: Why adding such logic for track2?

Comment (bebound) [reply src/azure-cli/azure/cli/command_modules/storage/util.py:27]: This logic is used to exclude directory as copy directory will raise an error. It seems like not all file has `content_md5`, I'll use try-except instead.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**